### PR TITLE
Update release script to match `uv`

### PR DESCRIPTION
**Pull request by zanieb**
See https://github.com/astral-sh/uv/pull/3764

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+cd "$project_root"
+uv tool run --from rooster-blue --isolated -- rooster release "$@"
+
+echo "Updating lockfile..."
+cargo update -p ruff
+
+echo "Generating contributors list..."
+echo ""
+echo ""
+rooster contributors --quiet
```
Comment (T-256) [inline scripts/release.sh:20]:
```suggestion
uv tool run --from rooster-blue --isolated -- rooster contributors --quiet
```

Comment (zanieb) [reply scripts/release.sh:20]: Ah thanks — I dropped the contributors display over in `uv`.

Maybe we should merge this one after #9559

**Review by charliermarsh — APPROVED** [review]

File: scripts/release.sh (hunk 1 above)
Comment (T-256) [reply scripts/release.sh]: Merged

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -331,7 +331,7 @@ even patch releases may contain [non-backwards-compatible changes](https://semve
 ### Creating a new release
 
 1. Install `uv`: `curl -LsSf https://astral.sh/uv/install.sh | sh`
-1. Run `./scripts/release/bump.sh`; this command will:
+1. Run `./scripts/release.sh`; this command will:
     - Generate a temporary virtual environment with `rooster`
```
Comment (T-256) [inline CONTRIBUTING.md:335]:
```suggestion
```
it is cached in user's data dir now.

Comment (zanieb) [reply CONTRIBUTING.md:335]: It's technically not even cached, idk if this matters though.

**Review by MichaReiser — APPROVED** [review]
Thanks. We probably need to update our contributing guidelines to mention how to install uv

**Pull request merged**